packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -115,25 +115,28 @@ export const withBlockBindingSupport = createHigherOrderComponent(
 
 					const keptAttributes = { ...nextAttributes };
 
-					for ( const [
-						attributeName,
-						boundAttribute,
-					] of Object.entries( bindings ) ) {
```

Comment:
Could you leave an inline comment on why you loop over the next attributes instead of bindings?

Reply:
Added in [this commit](https://github.com/WordPress/gutenberg/pull/60721/commits/1aa942e58054f713acd8bdcd2e6224ff8479119d). Let me know if that's not enough 🙂